client/src/panels/DataViewer.ts

```diff
@@ -4,31 +4,49 @@
 import { Uri } from "vscode";
 import PaginatedResultSet from "../components/LibraryNavigator/PaginatedResultSet";
 import { TableData } from "../components/LibraryNavigator/types";
+import { Column } from "../connection/rest/api/compute";
 import { WebView } from "./WebviewManager";
 
 class DataViewer extends WebView {
   private _uid: string;
   private _extensionUri: Uri;
   private _paginator: PaginatedResultSet<TableData>;
+  private _fetchColumns: () => Column[];
 
   public constructor(
     extensionUri: Uri,
     uid: string,
     paginator: PaginatedResultSet<TableData>,
+    fetchColumns: () => Column[],
   ) {
     super();
     this._uid = uid;
     this._extensionUri = extensionUri;
     this._paginator = paginator;
+    this._fetchColumns = fetchColumns;
   }
 
   public render(): WebView {
+    const policies = [
+      `default-src 'none';`,
+      `font-src ${this.panel.webview.cspSource} data:;`,
+      `img-src ${this.panel.webview.cspSource} data:;`,
+      `script-src ${this.panel.webview.cspSource};`,
+      `style-src ${this.panel.webview.cspSource};`,
+    ];
     this.panel.webview.html = `
       <!DOCTYPE html>
       <html lang="en">
         <head>
           <meta charset="UTF-8" />
           <meta name="viewport" content="width=device-width, initial-scale=1.0" />
+          <meta http-equiv="Content-Security-Policy" content="${policies.join(
+            " ",
+          )}" />
+          <link rel="stylesheet" href="${this.webviewUri(
+            this._extensionUri,
+            "DataViewer.css",
+          )}">
           <title>${this._uid}</title>
         </head>
         <body>
```

Comment:
I just happened to notice that there's a `<vscode-data-grid>` div. Is it still needed?

Reply:
That's now been removed